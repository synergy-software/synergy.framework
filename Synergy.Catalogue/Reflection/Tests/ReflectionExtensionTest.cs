using System;
using System.Collections.Generic;
using ApprovalTests;
using Xunit;

namespace Synergy.Catalogue.Reflection.Tests
{
    public class ReflectionExtensionTest
    {
        [Fact]
        public void Test()
        {
            var types = new List<Type>()
            {
                typeof(object),
                typeof(string),
                
                typeof(byte),
                typeof(sbyte),
                typeof(short),
                typeof(ushort),
                typeof(int),
                typeof(uint),
                typeof(long),
                typeof(ulong),
                typeof(char),
                typeof(float),
                typeof(double),
                typeof(decimal),
                typeof(bool),
                
                typeof(byte?),
                typeof(sbyte?),
                typeof(short?),
                typeof(ushort?),
                typeof(int?),
                typeof(uint?),
                typeof(long?),
                typeof(ulong?),
                typeof(char?),
                typeof(float?),
                typeof(double?),
                typeof(decimal?),
                typeof(bool?),
                
                typeof(DateTime),
                typeof(DateTimeOffset),
                typeof(List<int>),
                typeof(int[]),
                typeof(Dictionary<string, long>),
            };

            Approvals.VerifyAll(types, type => $"{type} => {type.GetFriendlyTypeName()}");
        }
    }
}