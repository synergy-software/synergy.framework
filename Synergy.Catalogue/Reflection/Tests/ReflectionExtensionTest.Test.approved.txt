System.Object => object
System.String => string
System.Byte => byte
System.SByte => sbyte
System.Int16 => short
System.UInt16 => ushort
System.Int32 => int
System.UInt32 => uint
System.Int64 => long
System.UInt64 => ulong
System.Char => char
System.Single => float
System.Double => double
System.Decimal => decimal
System.Boolean => bool
System.Nullable`1[System.Byte] => byte?
System.Nullable`1[System.SByte] => sbyte?
System.Nullable`1[System.Int16] => short?
System.Nullable`1[System.UInt16] => ushort?
System.Nullable`1[System.Int32] => int?
System.Nullable`1[System.UInt32] => uint?
System.Nullable`1[System.Int64] => long?
System.Nullable`1[System.UInt64] => ulong?
System.Nullable`1[System.Char] => char?
System.Nullable`1[System.Single] => float?
System.Nullable`1[System.Double] => double?
System.Nullable`1[System.Decimal] => decimal?
System.Nullable`1[System.Boolean] => bool?
System.DateTime => DateTime
System.DateTimeOffset => DateTimeOffset
System.Collections.Generic.List`1[System.Int32] => List<int>
System.Int32[] => int[]
System.Collections.Generic.Dictionary`2[System.String,System.Int64] => Dictionary<string, long>
